source/addonStore/dataManager.py

```diff
@@ -183,6 +183,7 @@ def _getCachedAddonData(self, cacheFilePath: str) -> Optional[CachedAddonsModel]
 			return None
 		try:
 			data = cacheData["data"]
+			cachedAddonData = _createStoreCollectionFromJson(data)

```

Comment:
I'd like to see better exception handling in `_createStoreCollectionFromJson`. In particular, if we get a `JSONEncodeError` or `KeyError` due to malformed data, invalidate the cache.

Reply:
Isn't that already occurring here for `KeyError`? Also I think you mean `JSONDecodeError`

Reply:
except (KeyError, JSONDecodeError):

Reply:
please change the exception handling below on line 189 to include JSONDecodeError

Reply:
@seanbudd , I've included JSONDecodeError in the exception as you requested.

Reply:
have you tested this? It doesnt seem like `JSONDecodeError` is imported from anywhere

Reply:
No, sorry, I haven't tested it and in fact VS Code is raising a warning. I cannot test this now. I'll try to import JSONDecodeError

Reply:
@seanbudd , I've imported JSONDecodeError and VS Code doesn't show warnings.

Reply:
please let me know when you have tested the code - we do not review or merge untested code